[LiveComponent][RFC] Simple file upload handling (#299)
symfony/ux: src/LiveComponent/src/Resources/doc/index.rst

        }
    }

Actions and file uploads
~~~~~~~~~~~~~~~~~~~~~~~~

.. versionadded:: 2.2

>     The ability to pass arguments to actions was added in version 2.2.

Reviewer: This needs to be updated ;)

> If you want live component to track and send files you first need

Reviewer: We will probably need to first explain that the simplest thing to do with file uploads is... nothing? Let them submit with the form like normal.

THEN we can say: but if you want to upload the file via live components, that's possible and here is how.

> to mark file upload inputs as ``file`` Stimulus target.::

Reviewer: ```suggestion
to mark file upload inputs as ``file`` Stimulus target:
```

(only double `::` right PHP blocks... and then you can omit the `.. code-block:: php`

.. code-block:: twig

    <input type="file" {{ stimulus_target('live', 'file') }} name="my_file" />

> Then, when defining action you need to use special ``files(name)`` modifier.::

Reviewer: ```suggestion
Next, upload the file to a custom `LiveAction` using the `upload_files` modifier:
```

.. code-block:: twig

    <div
        data-action="change->live#action"

>         data-action-name="prevent|files(my_file)|upload"

Reviewer: ```suggestion
        data-action-name="upload_files(my_file)|handleMyFileUpload"
```

So, I changed a few things:

A) No `prevent`. Not needed here, and I think it takes away focus.
B) I like `upload_files` as the modifier name. It feels like a "verb", which "feels" correct here. I want to "trigger the action of uploading"
C) Made the live action name super explicit that it is a custom thing for this example. At first, I was thinking `|upload` was another modifier 🙃

        <input type="file" {{ stimulus_target('live', 'file') }} name="my_file" />
    </div>

> This will send files from ``my_file`` file input. When used without argument

Reviewer: ```suggestion
This will upload the file from the ``my_file`` file input. If you use `upload_files` without any arguments,
it would upload all files that have the `file` target.
```

it would send all files from all ``file`` targets of the controller.

> If you want to send multiple files from a single input remember to suffix its' name

Reviewer: ```suggestion
If you want to send multiple files from a single input, remember to suffix its name
```

> with ``[]`` - both in HTML name attribute and ``files`` modifier argument.

Reviewer: ```suggestion
with ``[]`` - both in the HTML name attribute and the ``upload_files`` modifier argument - e.g. ``upload_files('photos[]')``.
```

> 

Reviewer: The missing piece (which I'm guessing you were waiting on to get more feedback) is the PHP logic. We're going to need a really nice documentation example of how the `LiveAction` looks, including executing validation (and re-rendering with an error), processing the file upload (i.e. moving it somewhere) and setting the final filename (e.g. `tulips.jpg`) from that process onto some other `LiveAction` property... or setting it into an entity and saving immediately.
